### PR TITLE
RUBY CAPSTONE PROJECT: Catalog of things

### PR DESCRIPTION
**Pull request by ngalamac**
In this project, the following was implemented:
1. Created classes from UML diagram in separate files called:
- item.rb as base class file which has a many to 1 relationship with classes in:
- genre.rb
- author.rb
- source.rb
- label.rb
- Also we created child classes inheriting from item class as defined in:
- book.rb
- music_album.rb
- game.rb
         In all of these classes, the corresponding attributes and methods were implemented
2. Created test files in a folder called spec, which tests each unit in the various classes using RSpec 
3. Implemented a user interface to interact with users
4. created app.rb and main.rb as entry point
5. created a schema.sql file that has the SQL commands for the database
6. Documented the project, including a video demo link in the project README.md file

### CODE REVIEW

File: storage/books.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
+[
```
Comment (ShadyShawkat) [inline storage/books.json:1]: - [ ] Please make sure that the app still works properly when the JSON files don't exist. If the storage directory doesn't exist, the following error appears. Kindly fix the error and make sure that the data is stored in a new JSON file if there isn't one.

image

File: database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  label_id INT NOT NULL,
+  multiplayer BOOLEAN NOT NULL,
+  last_played_at DATE NOT NULL,
+  publish_date DATE NOT NULL,
+  archived BOOLEAN NOT NULL,
+  PRIMARY KEY (id),
+  CONSTRAINT fk_genre FOREIGN KEY (genre_id) REFERENCES genres (id),
+  CONSTRAINT fk_author FOREIGN KEY (author_id) REFERENCES authors (id),
+  CONSTRAINT fk_label FOREIGN KEY (label_id) REFERENCES labels (id)
+);
```
Comment (ShadyShawkat) [inline database/schema.sql:79]: - [ ] Kindly make sure to create indexes for each of the created foreign keys.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**